### PR TITLE
os: timing: polish doxygen

### PR DESCRIPTION
**Pull request by dcpleung**
* Moves the architecture specific timing measurement APIs under the timing measurement APIs group.
* Add SoC and board specific API groups.
* Document each SoC and board specific API so the doc shows up for them.

### CODE REVIEW

**Review by andyross — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

File: include/zephyr/sys/arch_interface.h [hunk 1]
```diff
@@ -1054,7 +1054,13 @@ int arch_gdb_remove_breakpoint(struct gdb_ctx *ctx, uint8_t type,
 #include <zephyr/timing/types.h>
 
 /**
- * @ingroup arch-timing
+ * @brief Arch specific Timing Measurement APIs
+ * @defgroup timing_api_arch Arch specific Timing Measurement APIs
```
Comment (kartben) [inline include/zephyr/sys/arch_interface.h:1058]: do you want to add this `doxygengroup::` to Architecture Porting Guide / Timing? Right now it's now showing all the API's you moved under this new group anymore.

File: include/zephyr/timing/timing.h [hunk 2]
```diff
@@ -14,33 +14,246 @@
...
 
+/**
+ * @brief Timing Measurement APIs
+ * @defgroup timing_api Timing Measurement APIs
+ * @ingroup os_services
+ */
+
+/**
+ * @brief SoC specific Timing Measurement APIs
+ * @defgroup timing_api_soc SoC specific Timing Measurement APIs
```
Comment (kartben) [inline include/zephyr/timing/timing.h:25]: would it benefit from being mounted somewhere in the Sphinx doc? Probably the "Executing Time Functions" page?

File: include/zephyr/sys/arch_interface.h [hunk 3]
```diff
@@ -1098,17 +1104,25 @@ void arch_timing_stop(void);
 /**
  * @brief Return timing counter.
  *
+ * @parblock
```
Comment (kartben) [inline include/zephyr/sys/arch_interface.h:1107]: I am not sure the added parblocks are necessary and making any difference?

**Review by kartben — CHANGES_REQUESTED** [review]

File: include/zephyr/sys/arch_interface.h (hunk 3 above)
Comment (dcpleung) [reply include/zephyr/sys/arch_interface.h:1107]: These keeps the note sections separated or else both would be merged together as one big note block.

File: include/zephyr/timing/timing.h [hunk 4]
```diff
@@ -14,33 +14,246 @@
 extern "C" {
 #endif
 
+/**
+ * @brief Timing Measurement APIs
+ * @defgroup timing_api Timing Measurement APIs
+ * @ingroup os_services
+ */
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply include/zephyr/timing/timing.h:28]: Added.

File: include/zephyr/sys/arch_interface.h (hunk 1 above)
Comment (dcpleung) [reply include/zephyr/sys/arch_interface.h:1058]: The timing functions are not strictly required when porting to a new architecture, as long as you don't enable `CONFIG_TIMING_FUNCTIONS`. So I am hesitant in adding to that doc, as it seems to convey the idea that these are required.

File: include/zephyr/timing/timing.h (hunk 2 above)
Comment (dcpleung) [reply include/zephyr/timing/timing.h:32]: I totally forgot about that page... added these groups to that doc.

File: include/zephyr/sys/arch_interface.h (hunk 3 above)
Comment (kartben) [reply include/zephyr/sys/arch_interface.h:1107]: Oh you're quite right. I must have done something wrong when I thought I was testing how it looked on /latest and had the impression it was looking just the same

**Review by peter-mitsis — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**